src/impl/apis/data_api_impl.py

```diff
@@ -258,35 +259,152 @@ async def get_time_range(
     return time_range
 
 
+async def get_query_attributes(query_parameters, token_model, pool):
+    """
+    Gets the metadata + query parameters
+    """
+    category = await get_category(query_parameters.category_id, token_model=token_model, pool=pool)
+    if category.type not in ["single_location", "flow"]:
+        raise HTTPException(
+            status_code=HTTPStatus.NOT_FOUND,
+            detail="Could not find category",
```

Comment:
I think this is more like a 500 error really, so log the reason, return a generic-ish 500?